pkg/network/setup/configstatecache_test.go

```diff
@@ -0,0 +1,126 @@
+/*
+* This file is part of the KubeVirt project
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* Copyright 2023 Red Hat, Inc.
+*
+ */
+
+package network_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	network "kubevirt.io/kubevirt/pkg/network/setup"
+
+	dutils "kubevirt.io/kubevirt/pkg/ephemeral-disk-utils"
+
+	"kubevirt.io/kubevirt/pkg/network/cache"
+)
+
+var _ = Describe("config state cache", func() {
+	const (
+		uid     = "123"
+		testNet = "testnet"
+	)
+	var (
+		configStateCache network.ConfigStateCache
+		baseCacheCreator tempCacheCreator
+		ifacesMap        map[string]cache.PodIfaceState
+	)
+
+	BeforeEach(func() {
+		dutils.MockDefaultOwnershipManager()
+		ifacesMap = map[string]cache.PodIfaceState{}
+		configStateCache = network.NewConfigStateCacheWithStateMap(uid, &baseCacheCreator, ifacesMap)
+	})
+	AfterEach(func() {
+		Expect(baseCacheCreator.New("").Delete()).To(Succeed())
+	})
+
+	Context("read", func() {
+		It("from an empty cache", func() {
+			state, err := configStateCache.Read(testNet)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(state).To(Equal(cache.PodIfaceNetworkPreparationPending))
+		})
+		It("state stored only in the file cache", func() {
+			podIfaceCacheData := &cache.PodIfaceCacheData{State: cache.PodIfaceNetworkPreparationStarted}
+			Expect(cache.WritePodInterfaceCache(&baseCacheCreator, uid, testNet, podIfaceCacheData)).To(Succeed())
+
+			state, err := configStateCache.Read(testNet)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(state).To(Equal(cache.PodIfaceNetworkPreparationStarted))
+
+			Expect(ifacesMap[testNet]).To(Equal(cache.PodIfaceNetworkPreparationStarted))
+		})
+		It("state stored only in file cache while the memory cache is not empty", func() {
```

Comment:
It may happen after virt-handler restart. All the old interfaces will be in the file only, the newly hotplugged interfaces will be in both memory and the file. This test makes sure that once "old" interface that is stored in the file only is read, the memory cach is updated as well.

Reply:
Well, it is not overriding, adding. If the volatile.caxhe has key A and the persisted cache has keys A and B. When reading B it should be added to the volatile cache.